From 3dc5f008af8b726fd8cb1090b0a2056edb62876d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gy=C3=B6rgy=20Ball=C3=B3?= <ballogyor@gmail.com>
Date: Tue, 28 Jul 2015 03:27:32 +0200
Subject: [PATCH] removable-drives applet: don't hardcode Nemo

Use Gio to open the default file manager at 'computer:///'.
---
 .../share/cinnamon/applets/removable-drives@cinnamon.org/applet.js  | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js
index b3143de..8fb8f92 100644
--- a/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js
@@ -1,6 +1,6 @@
 const Lang = imports.lang;
 const St = imports.gi.St;
-const Cinnamon = imports.gi.Cinnamon;
+const Gio = imports.gi.Gio;
 const Applet = imports.ui.applet;
 const Main = imports.ui.main;
 const PopupMenu = imports.ui.popupMenu;
@@ -63,9 +63,7 @@ MyApplet.prototype = {
 
             this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
             this.menu.addAction(_("Open file manager"), function(event) {
-                let appSystem = Cinnamon.AppSystem.get_default();
-                let app = appSystem.lookup_app('nemo.desktop');
-                app.activate_full(-1, event.get_time());
+                Gio.app_info_launch_default_for_uri("computer:///", null);
             });     
             
             Main.placesManager.connect('mounts-updated', Lang.bind(this, this._update));
